core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -647,8 +647,14 @@ class KafkaController(val config: KafkaConfig,
         info(s"Skipping reassignment of $tp since the topic is currently being deleted")
         new ApiError(Errors.UNKNOWN_TOPIC_OR_PARTITION, "The partition does not exist.")
       } else {
-        val assignedReplicas = controllerContext.partitionReplicaAssignment(tp)
-        if (assignedReplicas.nonEmpty) {
+        val assignment = controllerContext.partitionFullReplicaAssignment(tp)
+        if (assignment == ReplicaAssignment.empty) {
+          new ApiError(Errors.UNKNOWN_TOPIC_OR_PARTITION, "The partition does not exist.")
+        } else if (assignment == reassignment) {
```

Comment:
It seems like the thing to agree on is the final state, right?  This comparison is taking into account the current replica set which may change over the course of the reassignment...

Reply:
I don't think there's a difference. The reassignment here is computed with respect to the current assignment. So if the target states match, then so will these.

Reply:
I ended up reverting this change because the controller integration tests were failing. I'll consider opening a separate jira for the problem of avoiding the epoch bump when the reassignment does not change.